Formula/v/vulkancapsviewer.rb

```diff
@@ -0,0 +1,30 @@
+class Vulkancapsviewer < Formula
+  desc "Client application to display hardware implementation details for Vulkan GPUs"
+  homepage "https://github.com/SaschaWillems/VulkanCapsViewer"
+  url "https://github.com/SaschaWillems/VulkanCapsViewer.git", tag: "3.32", revision: "3024e3148a7b47706b6d259f8e0ad0a2aa651330"
+  license "LGPL-3.0-or-later"
+
+  depends_on "cmake" => :build
+  depends_on "vulkan-headers" => :build
+  depends_on "qt@5"
+  depends_on "vulkan-loader"
+
+  def install
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args,
+                    "-DQt5_DIR=#{Formula["qt5"].opt_lib}/cmake/Qt5",
+                    "-DVULKAN_HEADERS_INSTALL_DIR=#{Formula["vulkan-headers"].opt_prefix}",
+                    "-DVULKAN_LOADER_INSTALL_DIR=#{Formula["vulkan-loader"].opt_prefix}"
+    system "cmake", "--build", "build"
+    if OS.mac?
+      bin.install "build/vulkanCapsViewer.app/Contents/MacOS/vulkanCapsViewer"
+    else
+      bin.install "build/vulkanCapsViewer"
+    end
+  end
+
+  test do
+    with_env("QT_QPA_PLATFORM" => "offscreen") do
+      system "#{bin}/vulkanCapsViewer", "--help"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
It's a GUI application. Looking at it, sure there is more stuff we can test.

Reply:
There are upstream bugs which prevent any kind of testing beyond basic "--help" level stuff.
See https://github.com/SaschaWillems/VulkanCapsViewer/issues/192

I thought about forcing the application to use mock vulkan driver, but that doesn't work either.